### PR TITLE
Added config file and support

### PR DESCRIPTION
**Pull request by davidronk**

### CODE REVIEW

File: example_config.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+squish_paths:
+  - '/Volumes/storage/videos/sports'
+  - '/Volumes/storage/videos/movies'
+  - '/Volumes/storage/videos/tv.shows/'
+rename_ts_paths:
+  - '/Volumes/storage/videos/tv.shows/'
+  - '/Volumes/storage/videos/sports'
+log_loc: "logs"
+processed_loc: "processed"
```
Comment (benronk) [inline example_config.yml:9]: Why not specify this like the other paths  `/Volumes/storage/videos/processed`. And why the double quotes?

File: process.rb [hunk 2]
```diff
@@ -1,39 +1,39 @@
...
   Dir[ File.join(path, '**', '*') ]
     .reject { |f| File.directory? f}
-    .reject { |f| $EXCLUDED_DIRS.any? { |p| File.dirname(f).downcase.include? p.downcase } }
+    .reject { |f| ($config["excluded_dirs"] || []).any? { |p| File.dirname(f).downcase.include? p.downcase } }
     .select { |f| $PROCESS_FILE_TAGS.any? { |p| File.basename(f).downcase.include? p.downcase } }
 end
 
 class TranscodableFile
-  def initialize(file)
+  def initialize(file, path)
```
Comment (benronk) [inline process.rb:28]: Why pass the path here? You can get the path from the file variable `File.dirname(file)`

File: process.rb [hunk 3]
```diff
@@ -48,7 +48,9 @@ def transcode_file
 
   # Remove the process tag and change path to videos/processed
   def moveto_file
-    @base_file.sub('.'+process_tag, '').sub('videos', 'videos/processed')
+    last_dir_in_path = Pathname(@base_path).each_filename.to_a.last
```
Comment (benronk) [inline process.rb:51]: `last_dir_in_path = Pathname(File.dirname(@base_file)).each_filename.to_a.last`

File: process.rb [hunk 4]
```diff
@@ -48,7 +48,9 @@ def transcode_file
...
+    file_wo_path = @base_file.sub('.'+process_tag, '').sub(@base_path, '')
```
Comment (benronk) [inline process.rb:52]: `file_wo_path = File.basename(@base_file.sub('.'+process_tag, ''))`

File: example_config.yml (hunk 1 above)
Comment (davidronk) [reply example_config.yml:9]: Double quotes were just habit; it doesn't matter here.

I left it that way to show that relative paths work.  `processed` would put the processed files in a directory next to the script.  It also works as `/Volumes/storage/videos/processed`.

File: process.rb (hunk 2 above)
Comment (davidronk) [reply process.rb:28]: It's not the file's path it's the base path set in `squish_paths`.  So when it puts it in `processed` it removes what was specified in `squish_paths`.

Comment (davidronk) [reply process.rb:28]: Basically it should put files exactly where it did before 🤞🏻 🙂

**Pull request merged**